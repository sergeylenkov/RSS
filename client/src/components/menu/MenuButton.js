import React from 'react';
import { Icon, Icons } from '../Icon.js';

import styles from './MenuButton.module.css';

export class MenuButton extends React.Component {
    render() {
        const feed = this.props.feed;
        const isEdited = this.props.isEdited;

        let className = styles.container;

        if (isEdited) {
            className += ` ${styles.edited}`;

            return (
                <button className={className}>
                    <div className={styles.icon} style={{ backgroundImage: `url(${feed.icon})` }}></div>
                    <div className={styles.label}>{feed.title}</div>
                    <button className={styles.deleteButton} onClick={() => this.onDelete()}><Icon svg={Icons.trash}/></button>                    
                </button>
            );
        } else {
            if (feed.count === 0) {
                className += ` ${styles.empty}`;
            }
    
            if (this.props.isSelected) {
                className += ` ${styles.selected}`;
            }

            return (
                <button className={className} onClick={() => this.onClick()}>
                    <div className={styles.icon} style={{ backgroundImage: `url(${feed.icon})` }}></div>
                    <div className={styles.label}>{feed.title}</div>
                    <div className={styles.counter}>{feed.count}</div>
                </button>
            );        
        }
    }

    onClick() {
        this.props.onClick(this.props.feed.id);
    }

    onDelete() {
        this.props.onDelete(this.props.feed.id);
    }
}