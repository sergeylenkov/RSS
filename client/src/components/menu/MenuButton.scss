.menu-button {
  display: flex;
  flex-direction: row;
  flex-shrink: 0;
  height: 100%;
  margin: 0 20px 0 0;
  cursor: pointer;

  &__label {
    margin: auto 0;
    font-size: 14px;
    color: #000000;

    &-selected {
      font-weight: bold;
    }
  }

  &__counter {
    margin: auto 0 auto 10px;
    padding: 2px 5px;
    font-size: 11px;
    border-radius: 2px;
    color: #ffffff;
    background-color: #4da0da;
  }
}

/*
* theme
*/

.dark .menu-button {
  &__label {
    color: #bfbfbf;

    &_selected {
      color: #f3f3f3;
    }
  }

  &__counter {
    color: #ffffff;
    background-color: #4da0da;
  }
}
