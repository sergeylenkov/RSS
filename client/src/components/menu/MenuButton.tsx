import React, { useCallback } from 'react';
import { Bem } from '../../utils/bem';

import './MenuButton.scss';

const block = new Bem('menu-button');

interface MenuButtonProps {
  isSelected: boolean;
  count: number;
  title: string;
  onClick: () => void;
}

function MenuButton({ isSelected, count, title, onClick } : MenuButtonProps): JSX.Element {
  const onClickHandle = useCallback(() => {
    if (!isSelected) {
      onClick();
    }
  }, [isSelected]);

  const blockClass = block.toString();
  const labelClass = block.getElement('label').addModifier(isSelected ? 'selected' : '').toString();
  const counterClass = block.getElement('counter').toString();

  return (
    <button className={blockClass} onClick={onClickHandle}>
      <div className={labelClass}>{title}</div>
      {isSelected && <div className={counterClass}>{count}</div>}
    </button>
  )
}

export default MenuButton;
