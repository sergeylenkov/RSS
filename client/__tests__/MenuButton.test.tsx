import React from 'react';
import renderer  from 'react-test-renderer';
import MenuButton from '../src/components/menu/MenuButton';

const action = () => { return };

test('MenuButton correct rendering', () => {
  const render = renderer.create(<MenuButton title={'Test'} isSelected count={10} onClick={action} />);
  expect(render).toMatchSnapshot();
});