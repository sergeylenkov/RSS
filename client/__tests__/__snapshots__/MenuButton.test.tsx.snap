// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButton correct rendering 1`] = `
<button
  className="menu-button"
  onClick={[Function]}
>
  <div
    className="menu-button__label menu-button__label_selected"
  >
    Test
  </div>
  <div
    className="menu-button__counter"
  >
    10
  </div>
</button>
`;
